uber/config.py

```diff
@@ -303,6 +303,28 @@ def REMAINING_BADGES(self):
     def ADMIN_ACCESS_SET(self):
         return sa.AdminAccount.access_set()
 
+    @request_cached_property
+    def EMAIL_APPROVED_IDENTS(self):
+        return self.get_approved_idents()
+
+    # TODO: only reason this is a separate method is because for unit tests we patch this method,
+    # and monkeypatch.setattr() doesn't like to patch EMAIL_APPROVED_IDENTS
+    # when it's decorated with @request_cached_property
+    def get_approved_idents(self):
+        with sa.Session() as session:
+            return {ae.ident for ae in session.query(sa.ApprovedEmail)}
```

Comment:
ok, I agree it kinda sucks to have to put this in Config when it's mostly unrelated to config settings.

the reason is that we are using ```@request_cached_property``` which will invoke the property once and cache it.  Normally the cache lasts until the end of the cherrypy request, however, because this code is in a thread which is not part of a cherrypy request, it stays cached forever and we have to manually unset it via ```with request_cached_context()```

So, these two properties need to be part of an instanced class, preferably a global one.  Config (c.WHATEVER) sort of fits that bill although we're starting to really bloat up that class.

I could also put this in some kind of separate EmailGlobals class that we instantiate maybe.  It would be about as messy.

Open to suggestions about how to fulfill all those requirements.  I think at some point we may need to do something major to re-organize Config and all the different separate things it's doing.